Dynavity/Dynavity/view/canvas/CanvasView.swift

```diff
@@ -1,21 +1,97 @@
 import SwiftUI
 
 struct CanvasView: View {
-    @ObservedObject var canvas: Canvas
+    @ObservedObject var viewModel = CanvasViewModel()
+
+    // For drag interactions
+    @State var portalPosition = CGPoint()
+    @State var dragOffset = CGSize()
+    @State var isDragging = false
+
+    // For zoom interactions
+    @State var zoomScale = CGFloat(1.0)
+    @State var initialZoomScale: CGFloat?
+    @State var initialPortalPosition: CGPoint?
 
     var body: some View {
-        ZStack {
-            Rectangle().fill(Color.blue)
-            CanvasElementMapView(elements: $canvas.canvasElements)
+        GeometryReader { geometry in
+            ZStack {
+                Rectangle()
+                CanvasElementMapView(elements: $viewModel.canvas.canvasElements)
+                    .scaleEffect(zoomScale)
+                    .offset(x: portalPosition.x + dragOffset.width,
+                            y: portalPosition.y + dragOffset.height)
+                    .animation(.easeIn)
+            }
+            .gesture(DragGesture()
```

Comment:
Could we extract the gestures out into their own variables? This will make `body` cleaner.